src/main/java/lab2/Lab2.java

```diff
@@ -0,0 +1,143 @@
+    var uri = URI.create("https://api.dicebear.com/9.x/%s/png?seed=%d".formatted(style, seed));
+    /*
+    6. uri is a reference type (URI)
+    2. URI.create() creates a URI by parsing the given string.
+    1. URI.create(): .create() is a class method of URI class (public final class)
+    1. .formatted is an instance method of the String
+    4. URI belongs to java.net
+     */
+    var request = HttpRequest.newBuilder(uri).build();
+    /*
+    6. request is a reference type (HttpRequest)
+    2. HttpRequest.newBuilder() creates an HttpRequest builder with the given URI
+    1. HttpRequest.newBuilder(): .newBuilder() is class method of the HttpRequest (public abstract class)
+    1. .build() is an instance method
+    4. HttpRequest belongs to java.net.http
+     */
+
+    // Send the request
+    try (var client = HttpClient.newHttpClient()) {
+        /*
+        6. client is a reference type (HttpClient)
+        2. HttpClient.newHttpClient() returns a new HttpClient with default settings
+        1. HttpClient.newHttpClient(): .newHttpClient() is a class method of the HttpClient (public abstract class)
+        4. HttpClient belongs to java.net.http
+         */
+        var response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+        /*
+        6. response is a reference (HttpResponse<InputStream>)
+        2. client.send() sends request and gets response
+        1. client.send(): .send() is an instance method
+        2. HttpResponse.BodyHandlers.ofInputStream() returns an image stream
+        1. HttpResponse.BodyHandlers is a class variable
+        1. HttpResponse.BodyHandlers.ofInputStream() is a class method
+        4. HttpResponse belongs to java.net.http
+         */
+        return response.body(); // 1. response.body() is an instance method and returns the body (InputStream)
+    }
+}
+
+void showAvatar(InputStream imageStream) { //7. the argument in showAvatar (InputStream imageStream) is a reference type
+    JFrame frame = new JFrame("PNG Viewer");
+    /*
+    6. frame is a reference type (JFrame)
+    3. JFrame has a constructor method and creates a new, initially invisible Frame with the specified title.
+    4. JFrame belongs to javax.swing
+    4. InputStream belongs to java.io
+     */
+    frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+    /*
+    1. frame.setDefaultCloseOperation() is an instance method
+    2. frame.setDefaultCloseOperation() sets the operation that will happen by default when the user initiates a "close" on this frame
+    1. JFrame.EXIT_ON_CLOSE) is class variable
+    4. JFrame belongs to javax.swing
+     */
+    frame.setResizable(false); //1. frame.setResizable() is an instance method and 2. sets whether this frame is resizable by the user
+    frame.setSize(200, 200); //1. frame.setSize() is an instance method and 2.resizes this component so that it has width and height
+    frame.getContentPane().setBackground(Color.BLACK);
```

Comment:
You didn't note whether the types of the arguments used in these method calls are primitive or reference types